File: frontend/lib/mongodb.ts
import { MongoClient, ServerApiVersion } from "mongodb";

const uri = process.env.MONGODB_URI
if (!uri) {
  throw new Error("MONGODB_URI is not set");
}


export const client = new MongoClient(uri, {
  serverApi: {
    version: ServerApiVersion.v1,
    strict: true,
    deprecationErrors: true,
  },
});

export async function connectDB() {
  try {
    if (!client.connect) return;
    await client.connect();
    await client.db("admin").command({ ping: 1 });
  } catch (error) {
    console.error("Error connecting to MongoDB:", error);
    throw error;
  }
}

File: backend/app/db/mongodb.py
import motor.motor_asyncio
import os
from dotenv import load_dotenv

load_dotenv()

MONGODB_URI = os.getenv("MONGODB_URI")
if not MONGODB_URI:
    raise ValueError("MONGODB_URI is not set in the .env file")
main_client = motor.motor_asyncio.AsyncIOMotorClient(MONGODB_URI)
main_db = main_client.get_database("test")

async def ping_mongodb():
    try:
        await main_client.admin.command("ping")
        print("Successfully connected to main")
    except Exception as e:
        print(f"Error connecting to MongoDB: {e}")
        raise e
